source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -420,27 +420,45 @@ void DecodingProcessorState::clearWatermark() {
 }
 
 void DecodingProcessorState::clearRouteCache(const CommonResponse& common_response) {
-  if (!common_response.clear_route_cache()) {
-    return;
-  }
   if (filter_.config().isUpstream()) {
-    filter_.stats().clear_route_cache_upstream_ignored_.inc();
-    ENVOY_LOG(debug, "NOT clearing route cache. The filter is in upstream filter chain.");
+    if (common_response.clear_route_cache()) {
```

Comment:
nit: `common_response.clear_route_cache()` appears 4 times  in this function alone.

Let's have a local bool variable and check that bool flag.